Do not link with libpthread - developed by Oracle
Not submitted upstream: specific to Solaris but no Solaris Makefile upstream

--- pigz-2.3.4/Makefile.orig	2017-03-02 13:22:22.763647319 +0000
+++ pigz-2.3.4/Makefile	2017-03-02 13:23:21.953773653 +0000
@@ -1,7 +1,7 @@
 CC=cc
 CFLAGS=-O3 -Wall -Wextra
 LDFLAGS=
-LIBS=-lm -lpthread -lz
+LIBS=-lm -lz
 ZOPFLI=zopfli/src/zopfli/
 # use gcc and gmake on Solaris
 
